src/main/java/application/DebuggerController.java

```diff
@@ -85,12 +83,12 @@ public MenuDebuggerContentResponseBean menuDebuggerContent(
         MenuSession menuSession = getMenuSessionFromBean(debuggerMenuRequest);
         runnerService.advanceSessionWithSelections(menuSession, debuggerMenuRequest.getSelections());
 
-        return new MenuDebuggerContentResponseBean(
+        return MenuController.setLocationNeeds(new MenuDebuggerContentResponseBean(
```

Comment:
Should have thought of this on my previous review, but I don't think the front end component of this will work for the debugger routes. They are handled in a separate `js` file outside of Marionette [here](https://github.com/dimagi/commcare-hq/blob/d07d1209f938dac9dbbd004cc800c02df89d23af/corehq/apps/cloudcare/static/cloudcare/js/debugger/debugger.js#L458-L467) so I don't think the callback hooks will work. IMO I don't think we need that working as a first pass